lib/widgets/editor/tree_widgets/waypoints_tree.dart

```diff
@@ -151,15 +159,32 @@ class _WaypointsTreeState extends State<WaypointsTree> {
           Tooltip(
             message: waypoint.isLocked ? 'Unlock' : 'Lock',
             waitDuration: const Duration(seconds: 1),
-            child: IconButton(
-              onPressed: () {
-                waypoint.isLocked = !waypoint.isLocked;
+            child: GestureDetector(
+              onTap: () {
+                setState(() {
+                  waypoint.isLocked = !waypoint.isLocked;
+                });
                 widget.onPathChanged?.call();
               },
-              icon: Icon(waypoint.isLocked ? Icons.lock : Icons.lock_open,
-                  color: colorScheme.onSurface),
+              child: AnimatedSwitcher(
+                duration: const Duration(milliseconds: 300),
+                transitionBuilder: (Widget child, Animation<double> animation) {
+                  return ScaleTransition(scale: animation, child: child);
+                },
+                child: Icon(
+                  waypoint.isLocked
+                      ? Icons.lock_rounded
+                      : Icons.lock_open_rounded,
+                  key: ValueKey<bool>(waypoint.isLocked),
+                  color: waypoint.isLocked
+                      ? colorScheme.primary
+                      : colorScheme.onSurface,
+                  size: 20,
+                ),
+              ),
```

Comment:
I'm pretty sure there's a way to use an animated icon with `IconButton`. `IconButton` takes in a `Widget` for the icon so you should be able to just pass in the `AnimatedSwitcher`. Using a gesture detector instead of an actual button sketches me out a little lol. It also has an ink well so it reacts to clicks and stuff.

Reply:
A bit of an oversight, its been updated now to use a normal icon button with an AnimatedSwitcher as the icon